internal/controllers/topology/cluster/patches/variables/variables.go

```diff
@@ -176,6 +192,22 @@ func Global(clusterTopology *clusterv1.Topology, cluster *clusterv1.Cluster) (Va
 			},
 		},
 	}
+	if cluster.Spec.ClusterNetwork != nil {
+		clusterNetworkIPFamily, err := cluster.GetIPFamily()
+		if err != nil {
+			return nil, err
+		}
+		builtin.Cluster.Network = &ClusterNetworkBuiltins{
+			ServiceDomain: &cluster.Spec.ClusterNetwork.ServiceDomain,
```

Comment:
Missed it before. Can we please only set ServiceDomain if the string is != "". 

Currently when serviceDomain is not set in YAML it leads to an empty `cluster.Spec.ClusterNetwork.ServiceDomain` field which then leads to a builtin variable set to an empty string.

Reply:
(You can reproduce via a new table test case which has ClusterNetwork set, but not ServiceDomain)